### PR TITLE
enable istio-sidecar-injector webhook in e2e pilot tests

### PR DESCRIPTION
**Pull request by ayj**
fixes https://github.com/istio/istio/issues/1463

### CODE REVIEW

File: prow/istio-pilot-e2e.sh [hunk 1]
```diff
@@ -48,12 +48,13 @@ fi
 
 source "${ROOT}/prow/cluster_lib.sh"
 
-trap delete_cluster EXIT
-create_cluster 'e2e-pilot'
+#trap delete_cluster EXIT
+#create_cluster 'e2e-pilot'
 
-ln -sf "${HOME}/.kube/config" ${ROOT}/pilot/pkg/kube/config
+KUBECONFIG=${ROOT}/pilot/pkg/kube/config
```
Comment (andraxylia) [inline prow/istio-pilot-e2e.sh]: prow job fails, it cannot find it:
	error	kubernetes configuration file "/home/bootstrap/go/src/istio.io/istio/pilot/pkg/kube/config" does not exist

File: prow/istio-pilot-e2e.sh [hunk 2]
```diff
@@ -56,4 +56,4 @@ HUB="gcr.io/istio-testing"
 
 cd ${GOPATH}/src/istio.io/istio
 ./bin/init.sh
-make -C "${ROOT}/pilot" e2etest HUB="${HUB}" TAG="${GIT_SHA}" TESTOPTS="-mixer=false -use-sidecar-injector=false -use-admission-webhook=false"
+make -C "${ROOT}/pilot" e2etest HUB="${HUB}" TAG="${GIT_SHA}" TESTOPTS="-mixer=false -use-sidecar-injector=true -use-admission-webhook=false
```
Comment (mattdelco) [inline prow/istio-pilot-e2e.sh]: The ending quote is missing.

It may be appropriate to make a similar change to 2 places in .circleci/config.yml

**Pull request merged**